core/src/apps/cardano/certificates.py

```diff
@@ -0,0 +1,272 @@
+
+        return (
+            certificate.type,
+            pool_parameters.pool_id,
+            pool_parameters.vrf_key_hash,
+            pool_parameters.pledge,
+            pool_parameters.cost,
+            cbor.Tagged(
+                30,
+                (
+                    pool_parameters.margin_numerator,
+                    pool_parameters.margin_denominator,
+                ),
+            ),
+            # this relies on pool_parameters.reward_account being validated beforehand
+            # in _validate_pool_parameters
+            get_address_bytes_unsafe(pool_parameters.reward_account),
+            _cborize_pool_owners(keychain, pool_parameters.owners),
+            _cborize_pool_relays(pool_parameters.relays),
+            _cborize_pool_metadata(pool_parameters.metadata),
+        )
+    else:
+        raise INVALID_CERTIFICATE
+
+
+def _validate_pool_parameters(
+    pool_parameters: CardanoPoolParametersType, protocol_magic: int, network_id: int
+) -> None:
+    if any(
+        [
+            not pool_parameters.pool_id
+            or len(pool_parameters.pool_id) != POOL_HASH_SIZE,
+            (
+                not pool_parameters.vrf_key_hash
+                or len(pool_parameters.vrf_key_hash) != VRF_KEY_HASH_SIZE
+            ),
+            not 0 <= pool_parameters.pledge < LOVELACE_MAX_SUPPLY,
+            not 0 <= pool_parameters.cost < LOVELACE_MAX_SUPPLY,
+            pool_parameters.margin_numerator <= 0,
+            pool_parameters.margin_denominator <= 0,
+            pool_parameters.margin_numerator > pool_parameters.margin_denominator,
+            (not pool_parameters.owners or len(pool_parameters.owners) == 0),
+        ]
+    ):
+        raise INVALID_CERTIFICATE
+
+    validate_reward_address(pool_parameters.reward_account, protocol_magic, network_id)
+
+    for pool_relay in pool_parameters.relays:
+        _validate_pool_relay(pool_relay)
+
+    _validate_pool_owners(pool_parameters.owners)
+
+    if pool_parameters.metadata:
+        _validate_pool_metadata(pool_parameters.metadata)
+
+
+def _validate_pool_owners(owners: List[CardanoPoolOwnerType]) -> None:
+    owners_as_path_count = 0
+    for owner in owners:
+        if any(
+            [
+                not owner.staking_key_hash and not owner.staking_key_path,
+                (
+                    owner.staking_key_hash
+                    and len(owner.staking_key_hash) != PUBLIC_KEY_HASH_SIZE
+                ),
+                (
+                    owner.staking_key_path
+                    and not SCHEMA_STAKING.match(owner.staking_key_path)
+                ),
+            ]
+        ):
+            raise INVALID_CERTIFICATE
+
+        if owner.staking_key_path:
+            owners_as_path_count += 1
+
+    if owners_as_path_count != 1:
+        raise INVALID_CERTIFICATE
+
+
+def _validate_pool_relay(pool_relay: CardanoPoolRelayParametersType) -> None:
+    if pool_relay.type not in [
```

Comment:
there should not be more relay types in the foreseeable future and even if they were it would almost certainly impact this validation logic. I agree the probuf enum check is enough